src/stores/shows/ShowsEffect.ts

```diff
@@ -11,34 +10,19 @@ export default class ShowsEffect {
 
   public static async requestShow(showId: string): Promise<ShowModel | HttpErrorResponseModel> {
     const endpoint: string = environment.api.shows.replace('{showId}', showId);
-    const response: AxiosResponse | HttpErrorResponseModel = await ShowsEffect._http.get(endpoint);
 
-    if (response instanceof HttpErrorResponseModel) {
-      return response;
-    }
-
-    return new ShowModel(response.data);
+    return ShowsEffect._http.getToModel<any>(ShowModel, endpoint);
```

Comment:
@ccheney how do you feel about having to do `private static _http = new HttpUtility();` in every Effect Class? Should I make HttpUtility a static class? So it would go from:

```
return ShowsEffect._http.getToModel(ShowModel, endpoint);
```
TO:
```
return HttpUtility.getToModel(ShowModel, endpoint);
```

I have this feeling that `HttpUtility` should not be a static class.

Reply:
@n8rzz

Reply:
Why do you feel it should not be a singleton?  Would this class maintain a state of some kind?